packages/react/src/components/hooks/useWebRTC.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+import { WebRTC } from "@signalwire/js";
+
+export default function useWebRTC(deviceEnabled: {
+  cameras: boolean;
+  microphones: boolean;
+  speakers: boolean;
+}) {
+  const [cameras, setCameras] = useState<MediaDeviceInfo[]>([]);
+  const [microphones, setMicrophones] = useState<MediaDeviceInfo[]>([]);
+  const [speakers, setSpeakers] = useState<MediaDeviceInfo[]>([]);
+
+  useEffect(() => {
+    async function setupList() {
+      if (deviceEnabled.cameras !== false) {
+        const cameras = await WebRTC.getCameraDevicesWithPermissions();
+        setCameras(Array.from(cameras));
+      }
+      if (deviceEnabled.microphones !== false) {
+        const microphones = await WebRTC.getMicrophoneDevicesWithPermissions();
+        setMicrophones(microphones);
+      }
+      if (deviceEnabled.speakers !== false) {
+        const speakers = await WebRTC.getSpeakerDevicesWithPermissions();
+        setSpeakers(speakers);
+      }
+    }
+    setupList();
+  }, []);
+
+  useEffect(() => {
+    let cameraWatcher: any, microphoneWatcher: any, speakerWatcher: any;
+    async function setupWatchers() {
+      if (deviceEnabled.cameras !== false) {
+        cameraWatcher = await WebRTC.createCameraDeviceWatcher();
+        cameraWatcher.on("changed", (x: any) => {
+          setCameras(x.devices);
+        });
+      }
+
+      if (deviceEnabled.microphones !== false) {
+        microphoneWatcher = await WebRTC.createMicrophoneDeviceWatcher();
+        microphoneWatcher.on("changed", (x: any) => {
+          setMicrophones(x.devices);
+        });
+      }
+
+      if (deviceEnabled.speakers !== false) {
+        speakerWatcher = await WebRTC.createSpeakerDeviceWatcher();
+        speakerWatcher.on("changed", (x: any) => {
+          setSpeakers(x.devices);
+        });
```

Comment:
same as above, these should be awaited in parallel unless there is a reason for doing it sequentially